landscaper/gardenlet/blueprint.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: Blueprint
+jsonSchema: "https://json-schema.org/draft/2019-09/schema" # required
+imports:
+  - name: seedCluster
+    targetType: landscaper.gardener.cloud/kubernetes-cluster
+
+  - name: gardenCluster
+    targetType: landscaper.gardener.cloud/kubernetes-cluster
+
+  - name: seedBackupCredentials
+    required: false
+    schema:
+      # provider specific credentials with varying properties
+      type: object
+
+  - name: imageVectorOverwrite
+    required: false
+    schema:
+      type: object
+
+  - name: componentImageVectorOverwrites
+    required: false
+    schema:
+      type: object
+
+  - name: deploymentConfiguration
+    required: false
+    schema:
+      type: object
+      properties:
+        replicaCount:
+          schema:
+            type: number
+        revisionHistoryLimit:
+          schema:
+            type: number
+        serviceAccountName:
+          schema:
+            type: string
+        resources:
+          schema:
+            type: object
+            properties:
+              requests:
+                type: object
+                properties:
+                  cpu:
+                    type: string
+                  memory:
+                    type: string
+              limits:
+                type: object
+                properties:
+                  cpu:
+                    type: string
+                  memory:
+                    type: string
+        podLabels:
+          schema:
+            type: object
+        podAnnotations:
+          schema:
+            type: object
+        additionalVolumes:
+          schema:
+            type: array
+            items:
+              type: object
+        additionalVolumeMounts:
+          schema:
+            type: array
+            items:
+              type: object
+        vpa:
+          schema:
+            type: bool
+
+  # check example/20-componentconfig-gardenlet.yaml for what can be configured here
+  - name: componentConfiguration
+    schema:
+      type: object
+
+# For what can be referenced in the deployExecutions, please see:
+# https://github.com/gardener/landscaper/blob/master/docs/usage/Blueprints.md
+deployExecutions:
+  - name: default
+    type: GoTemplate
+    template: |
+      deployItems:
+      - name: deploy
+        type: landscaper.gardener.cloud/container
+        config:
+          apiVersion: container.deployer.landscaper.gardener.cloud/v1alpha1
+          kind: ProviderConfiguration
+          blueprint:
+      {{ toYaml .blueprint | indent 6 }}
+
+          componentDescriptor:
+      {{ toYaml .componentDescriptorDef | indent 6 }}
```

Comment:
There is the documentation in [the landscaper repository](https://github.com/gardener/landscaper/tree/master/docs).
The goal is to have that automatically tested with [PR integration tests](https://github.com/gardener/gardener/pull/3757).

Reply:
So you confirm that this `indent 6` works as expected here?

Reply:
Yes in my tests it works, also verified with @schrodit . 
We can already see that this is hard / annoying to test whenever something changes in these definitions. Hence, we want to automatically validate their correctness on every PR against a gardenlet build from the commit.

I see this is a first step and I am sure that there will be problems/changes required. But for now we need to gain experience with it first.

Reply:
yes this works because it is a inlined template.
We could move the template to a separate file and improve readability (then the file is just like any other helm templated file).
This would mean that we would have the blueprint file and besides a `deployItem.tpl` file.

The downside is that you would have to maintain another file and that the imports the usage is split.
WDYT?